From 85c960319fa20f15d0bdec5a7bf0b4ca89fba14b Mon Sep 17 00:00:00 2001
From: David Gibson <dgibson@redhat.com>
Date: Mon, 15 Sep 2014 07:13:03 -0400
Subject: [virt] kvm/ppc: book3s - Ignore DABR register

Message-id: <1410765214-16377-48-git-send-email-dgibson@redhat.com>
Patchwork-id: 94735
O-Subject: [PATCH 47/78] KVM: PPC: Book3S: Ignore DABR register
Bugzilla: 1123145 1123133 1123367
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>

Signed-off-by: David Gibson <dgibson@redhat.com>

commit f35320288c5306ddbcb5ecac046b73519837299c
Author: Alexander Graf <agraf@suse.de>
Date:   Tue Jul 2 16:15:10 2013 +0200

    KVM: PPC: Book3S: Ignore DABR register

    We don't emulate breakpoints yet, so just ignore reads and writes
    to / from DABR.

    This fixes booting of more recent Linux guest kernels for me.

    Reported-by: Nello Martuscielli <ppc.addon@gmail.com>
    Tested-by: Nello Martuscielli <ppc.addon@gmail.com>
    Signed-off-by: Alexander Graf <agraf@suse.de>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kvm/book3s_emulate.c b/arch/powerpc/kvm/book3s_emulate.c
index 3bcb73b..45d0a80 100644
--- a/arch/powerpc/kvm/book3s_emulate.c
+++ b/arch/powerpc/kvm/book3s_emulate.c
@@ -454,6 +454,7 @@ int kvmppc_core_emulate_mtspr_pr(struct kvm_vcpu *vcpu, int sprn, ulong spr_val)
  case SPRN_PMC4_GEKKO:
  case SPRN_WPAR_GEKKO:
  case SPRN_MSSSR0:
+ case SPRN_DABR:
   break;
 unprivileged:
  default:
@@ -551,6 +552,7 @@ int kvmppc_core_emulate_mfspr_pr(struct kvm_vcpu *vcpu, int sprn, ulong *spr_val
  case SPRN_PMC4_GEKKO:
  case SPRN_WPAR_GEKKO:
  case SPRN_MSSSR0:
+ case SPRN_DABR:
   *spr_val = 0;
   break;
  default:
-- 
1.7.1